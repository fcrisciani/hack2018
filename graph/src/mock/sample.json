{
  "graph": [
    {
      "name": "service 1",
      "ip": "192.168.0.1",
      "connections": [
        {
          "total": 0
        },
        {
          "total": 2
        },
        {
          "total": 6
        },
        {
          "total": 9
        }
      ]
    },
    {
      "name": "service 2",
      "ip": "192.168.0.2",
      "connections": [
        {
          "total": 0
        },
        {
          "total": 0
        },
        {
          "total": 6
        },
        {
          "total": 1
        }
      ]
    },
    {
      "name": "service 3",
      "ip": "192.168.0.3",
      "connections": [
        {
          "total": 6
        },
        {
          "total": 0
        },
        {
          "total": 0
        },
        {
          "total": 1
        }
      ]
    },
    {
      "name": "service 4",
      "ip": "192.168.0.4",
      "connections": [
        {
          "total": 4
        },
        {
          "total": 0
        },
        {
          "total": 4
        },
        {
          "total": 0
        }
      ]
    }
  ]
}