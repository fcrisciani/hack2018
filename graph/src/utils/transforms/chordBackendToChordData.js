export default (graph) => {
  const matrix = [];
  const labels = []
  graph.forEach((serviceDesc) => {
    matrix.push([]);
    labels.push(serviceDesc.name);
  });
  graph.forEach((serviceDesc, i) => {
    serviceDesc.connections.forEach((connection, j) => {
      matrix[i][j] = connection.total;
    });
  })
  return {
    matrix,
    labels
  }
}
