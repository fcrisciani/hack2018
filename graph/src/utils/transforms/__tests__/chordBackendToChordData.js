import chordBackendToChordData from '../chordBackendToChordData';
import sample from './backendChordSample.json';

it('data transform', () => {
  expect(chordBackendToChordData(sample.graph)).toMatchSnapshot();
})