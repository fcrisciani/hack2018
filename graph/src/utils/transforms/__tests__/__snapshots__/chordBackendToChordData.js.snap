// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data transform 1`] = `
Object {
  "labels": Array [
    "service 1",
    "service 2",
    "service 3",
    "service 4",
  ],
  "matrix": Array [
    Array [
      0,
      2,
      6,
      9,
    ],
    Array [
      0,
      0,
      6,
      1,
    ],
    Array [
      6,
      0,
      0,
      1,
    ],
    Array [
      4,
      0,
      4,
      0,
    ],
  ],
}
`;
